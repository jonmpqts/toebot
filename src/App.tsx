import * as React from 'react';
import Board from './Board';

interface State {
  currentPlayer: 'X' | 'O';
  boardState: string;
}

class App extends React.Component<{}, State> {
  constructor(props: {}) {
    super(props);

    this.state = {
      currentPlayer: 'X',
      boardState: '.........',
    };

    this.handleSquareClick = this.handleSquareClick.bind(this);
  }

  handleSquareClick(id: number) {
    let newBoardState = this.state.boardState.split('');
    newBoardState[id] = this.state.currentPlayer;
    this.setState({
      currentPlayer: this.state.currentPlayer === 'X' ? 'O' : 'X',
      boardState: newBoardState.join(''),
    });
  }

  render() {
    return (
      <Board
        currentPlayer={this.state.currentPlayer}
        state={this.state.boardState}
        onSquareClick={this.handleSquareClick}
      />
    );
  }
}

export default App;
