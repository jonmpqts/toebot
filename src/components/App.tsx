import * as React from 'react';
import { connect } from 'react-redux';
import { Dispatch, Store } from 'redux';
import Board from './Board';
import { State, occupySquare, getCurrentPlayer, getBoardState } from '../store';

interface Props {
  currentPlayer: 'X' | 'O';
  boardState: string;
  onSquareClick: (id: number) => void;
}

export const App = ({ currentPlayer, boardState, onSquareClick }: Props) => {
  return (
    <Board
      currentPlayer={currentPlayer}
      state={boardState}
      onSquareClick={onSquareClick}
    />
  );
};

interface StateProps {
  currentPlayer: 'X' | 'O';
  boardState: string;
}

const mapStateToProps = (state: State) => ({
  currentPlayer: getCurrentPlayer(state),
  boardState: getBoardState(state)
});

interface DispatchProps {
  onSquareClick: (player: 'X' | 'O', squareNumber: number) => void;
}

const mapDispatchToProps = (dispatch: Dispatch<Store<State>>) => ({
  onSquareClick: (player: 'X' | 'O', squareNumber: number) =>
    dispatch(occupySquare(player, squareNumber))
});

interface OwnProps {}

const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps, ownProps: OwnProps) => ({
  ...ownProps,
  ...stateProps,
  ...dispatchProps,
  onSquareClick: (squareNumber: number) => {
    return dispatchProps.onSquareClick(stateProps.currentPlayer, squareNumber);
  }
});

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(App);
