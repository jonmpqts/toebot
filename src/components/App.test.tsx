import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { App } from './App';

it('renders without crashing', () => {
  const component = renderer.create(
    <App
      currentPlayer="X"
      boardState="........."
      winner={null}
      onSquareClick={() => { return; }}
    />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
