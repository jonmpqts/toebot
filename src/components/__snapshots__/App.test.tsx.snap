// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="board"
>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
  <button
    className="square open"
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </button>
</div>
`;
